pkg/services/datasources/lbac_rules.go

```diff
@@ -0,0 +1,24 @@
+package datasources
+
+import "github.com/grafana/grafana/pkg/components/simplejson"
+
+func CheckTeamHTTPHeadersDiff(currentJsonData *simplejson.Json, newJsonData *simplejson.Json) bool {
+	getEncodedString := func(jsonData *simplejson.Json, key string) string {
+		if jsonData == nil {
+			return ""
+		}
+		jsonValues, exists := jsonData.CheckGet(key)
+		if !exists {
+			return ""
+		}
+		val, _ := jsonValues.Encode()
+		return string(val)
+	}
+
+	currentTeamHTTPHeaders := getEncodedString(currentJsonData, "teamHttpHeaders")
+	newTeamHTTPHeaders := getEncodedString(newJsonData, "teamHttpHeaders")
+	if currentTeamHTTPHeaders == "" && newTeamHTTPHeaders == "" {
+		return false
+	}
+	return currentTeamHTTPHeaders != newTeamHTTPHeaders
```

Comment:
You are doing a string compare here, not an actual comparison of the json objects.

This means that if rules are formatted differently or in different orders, you'll get a diff. Is this ok?
`[ { rule 1 }, { rule 2 } ] != [ { rule 2 }, { rule 1 } ]`

Reply:
we do not need the check now that we instead override the previous rules